### ISSUE TITLE
Text size is too small in Windows

### ISSUE THREAD

**Issue by malwoden** [papyros/qml-material#230]
The following qml application: 

```qml
ApplicationWindow {
    visible: true
    initialPage: Page {
        id: page
        title: "DPI"

        tabs: ["tab1", "tab2", "a bit longer tab"]

        actions: [
            Action {
                name: "Settings"
                iconName: "action/settings"
            }
        ]
    }
}
```
Will result in the following Window:
![qml_material_windows_dpi](https://cloud.githubusercontent.com/assets/153354/8228020/86f6d81e-15a3-11e5-822f-b56c3dd7ae23.png)

This is at default Windows DPI settings (96 iirc). 

Units.pixelDensity is 2.8354450571988514 on Component Completed.

**Comment by ukoehler** [comment]
I verified the problem on Linux and Windows desktops. Looking at rschiang/material I also found the problem. In ApplicationWindow.qml you use Screen.pixelDensity in Component.onCompleted. The text size seem correct when using Screen.logicalPixelDensity. I get 72 and 96 pdi returned respectively.

Workaround: In any application add 
Component.onCompleted: {
        Units.pixelDensity = Qt.binding(function() { 
            return Screen.logicalPixelDensity
        });
}

**Comment by ukoehler** [comment]
Hmm, this is complicated. To get a better grip on it, I wrote a small QtQuick project showing 10 mm squares based on Screen.pixelDensity and Screen.logicalPixelDensity. The result:

Windows: use logicalPixelDensity
Linux: both are very similar and correct
Android: use pixelDensity

**Comment by nimaim** [comment]
Text appears incredibly small inside my Linux Mint environment as well. This is on a Retina screen though so it's to be expected, but the Macbook scales everything else correctly. What is the best way to handle this so that it will work across varying platforms? I think I saw another issue opened with this problem as well (marco's testing of resolution on different devices).

**Comment by marco-piccolino** [comment]
@nimaim I am still using a per-device fix (see https://gist.github.com/marco-piccolino/5a3f742474519abbe0d8).

I haven't investigated the issue deeply enough and probably the current Units code could be improved to behave better on the various platforms, but at least in some cases it's Qt getting wrong metrics from the devices/screens (see https://bugreports.qt.io/browse/QTBUG-46380 and suggested fix in there).

Also, I think @FiloSpaTeam suggested a native fix for Android that gets the density directly from the device.

**Comment by nimaim** [comment]
Thanks, I guess that per-device hack works for the time-being. Unfortunately, that does not account for every possible device but it's a start. I'm still new to QML but is the qml material demo here targeted towards an optimal screen resolution or platform (desktop/mobile/etc.)? I ran it on my 1440p screen at home and it looks very tiny but would have thought it'd scale.

**Comment by marco-piccolino** [comment]
@nimaim it is a part of the Papyros project (archlinux on desktop). As @iBeliever wrote in the readme, "It is completely cross platform, and runs on Linux, OS X, and Windows. It may also run on iOS and Android, though those platforms have not been tested and are not currently officially supported."
So it should display well on your machine as well.
What do you mean by tiny? window size or size of fonts/components?

**Comment by nimaim** [comment]
See the following (the form takes up like 5% of my whole screen, so obviously if you enlarge it, the form appears super small). Everything is small, window size AND fonts/components. Again, this is on a HiDPi screen so maybe this is normal.
![screenshot from 2015-09-22 09 48 45](https://cloud.githubusercontent.com/assets/9406197/10020277/05f1597a-6110-11e5-96ba-54e8b91076db.png)
![screenshot from 2015-09-22 09 49 59](https://cloud.githubusercontent.com/assets/9406197/10020278/060f01d2-6110-11e5-922a-07c1bd8f9efd.png)
![screenshot from 2015-09-22 09 52 03](https://cloud.githubusercontent.com/assets/9406197/10020279/0611c5ac-6110-11e5-8e65-f9058546efe0.png)

**Comment by nimaim** [comment]
Ok so the screens actually look normal on here :-/ But once you make it full screen, you can imagine how tiny they are.

**Comment by ukoehler** [comment]
Hi there,

really glad the issue has been taken up. It would be interesting to know how large a rectangle appears on the screen if you use  Screen.pixelDensity or Screen.logicalPixelDensity to calculate 1 inch or 1 cm.

Windows: Screen.logicalPixelDensity is correct
Linux: almost the same both visually correct
Android: Sreen.pixelDensity is correct (Moto G 1 Gen)

I posted to QtQuick forum 2 month ago, but got no help. This is used to setup the font sizes in qml-material.

Cheers

Uwe

**Comment by VirusKA** [comment]
This is a problem that needs to be fixed.

**Comment by iBeliever** [comment]
@VirusKA This _seems_ to be a mostly upstream Qt issue. However, I'm trying to come up with a solution that handles different OSes, and in addition, works on different screens. Unfortunately, that may require some C++ code, which we've been fortunate to have avoided so far.

**Comment by glenfordwilliams** [comment]
A c++ solution came to mind but I decided against when I thought of persons
using qt bindings in other languages eg pyside/pyqt

**Comment by VirusKA** [comment]
ah i see. I thought it was a Qt issue as well

**Comment by iBeliever** [comment]
@glenfordwilliams How would C++ be a problem for that use case? It'll be compiled as a separate module/plugin, so you should just be able to import the module as you do now. The only concern of using C++ that I know of is because I want to simplify the installation process. With C++, you'd have to explicitly compile and install the module instead of just copying it into your project as necessary.

**Comment by glenfordwilliams** [comment]
I was thinking along the lines of consistency

**Comment by nimaim** [comment]
@iBeliever I seriously think this is something that warrants some C++ code, unless we want to wait til its fixed in Qt upstream. I agree though, definitely complicates set up. Qml material is such a beautiful project but its a serious issue if the forms and components are not scaling properly.

**Comment by VirusKA** [comment]
any fix?

**Comment by VirusKA** [comment]
this has been fixed in Qt 5.6! the beta was released yesterday!

**Comment by ukoehler** [comment]
Great, will have to try that soon.

**Comment by malwoden** [comment]
![image](https://cloud.githubusercontent.com/assets/153354/12377493/1839c2e8-bd18-11e5-9554-115dd6a2cf3d.png)

As @VirusKA  says, this now appears fixed in 5.6 beta. Although arguably it has gone to far the other way though.

**Comment by iBelieve** [comment]
@malwoden We might need to disable our own code for Qt 5.6, as it might be duplicating the scaling of Qt 5.6. Also, there's a bug and the tooltip is extra tall; that's unrelated to this issue.

**Comment by iBelieve** [comment]
This is now fixed on Qt 5.6 and hopefully on previous versions as well. Please open a new issue if you're still facing problems. Thanks!
